provider/pkg/provider/helm_release.go

```diff
@@ -1099,12 +1122,22 @@ func (r *helmReleaseProvider) computeResourceNames(rel *Release, clientSet *clie
 		return nil, err
 	}
 
+	// compute the resource names
 	_, resourceNames, err := convertYAMLManifestToJSON(templ)
 	if err != nil {
 		return nil, err
 	}
+	result.resourceNames = resourceNames
+
+	// compute the checksum of the rendered output, to be able to detect changes
+	h := sha256.New()
```

Comment:
Encoding the SHA into state is an efficient way to check for diffs here. I think this will work well.

---

I'm not sure how big the manifests we are talking about are. If they are small, could we just stick the whole JSON blob in state? That way, the engine can provide high quality diffs to our users, instead of just showing that two random strings differ. 

Not a blocker if not viable.

Reply:
The size is arbitrarily large but I agree that the diff would be more useful.  That said, Helm keeps historical information and even has a `diff` plugin to show what's changed after it is applied. I'm not sure whether pulumi diff can show arbitrary information that isn't directly related to the properties.  I think we should consider a future enhancement with a larger time budget.

Reply:
There is an upper bound of 1.5 mb I think. Essentially the max size allowed for a k8s object which is ultimately defined by underlying etcd.